From f92090be9dbbe166dbb3143b539448773922efce Mon Sep 17 00:00:00 2001
From: Esteve Fernandez <esteve.fernandez@bonsairobotics.ai>
Date: Mon, 11 May 2026 09:40:11 +0100
Subject: [PATCH] devendor nlohmann_json

Signed-off-by: Esteve Fernandez <esteve.fernandez@bonsairobotics.ai>
---
 CMake/external_json.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMake/external_json.cmake b/CMake/external_json.cmake
index 74fd119..4ca5eaa 100644
--- a/CMake/external_json.cmake
+++ b/CMake/external_json.cmake
@@ -46,5 +46,7 @@ function(get_nlohmann_json)
 
 endfunction()
 
-# Trigger the build
-get_nlohmann_json()
+# Skip vendored json if already provided by system
+if(NOT TARGET nlohmann_json::nlohmann_json)
+    get_nlohmann_json()
+endif()
-- 
2.54.0

