From 25944c2bcbc8cc0775e743a4ccd7adb9f06e4bb9 Mon Sep 17 00:00:00 2001
From: Esteve Fernandez <esteve.fernandez@bonsairobotics.ai>
Date: Mon, 11 May 2026 10:12:46 +0100
Subject: [PATCH] fix pybind11 def_property_readonly keep_alive incompatibility

Signed-off-by: Esteve Fernandez <esteve.fernandez@bonsairobotics.ai>
---
 wrappers/python/pyrs_frame.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wrappers/python/pyrs_frame.cpp b/wrappers/python/pyrs_frame.cpp
index c5e07bd..ebf6bd7 100644
--- a/wrappers/python/pyrs_frame.cpp
+++ b/wrappers/python/pyrs_frame.cpp
@@ -144,7 +144,7 @@ void init_frame(py::module &m) {
         .def_property_readonly("frame_number", &rs2::frame::get_frame_number, "The frame number. Identical to calling get_frame_number.")
         .def("get_data_size", &rs2::frame::get_data_size, "Retrieve data size from frame handle.")
         .def("get_data", get_frame_data, "Retrieve data from the frame handle.", py::keep_alive<0, 1>())
-        .def_property_readonly("data", get_frame_data, "Data from the frame handle. Identical to calling get_data.", py::keep_alive<0, 1>())
+        .def_property_readonly("data", py::cpp_function(get_frame_data, py::keep_alive<0, 1>()), "Data from the frame handle. Identical to calling get_data.")
         .def("get_profile", &rs2::frame::get_profile, "Retrieve stream profile from frame handle.")
         .def_property_readonly("profile", &rs2::frame::get_profile, "Stream profile from frame handle. Identical to calling get_profile.")
         .def("keep", &rs2::frame::keep, "Keep the frame, otherwise if no refernce to the frame, the frame will be released.")
-- 
2.54.0

